metrics_block.py

```diff
@@ -54,6 +65,56 @@ def stop(self):
         if self._metrics_job is not None:
             self._metrics_job.cancel()
 
+    def timestamp(self):
+        '''returns the current system timestamp'''
+        return datetime.isoformat(datetime.utcnow())
+
+    @staticmethod
```

Comment:
The timestamp method right before this also doesn't use a self, I think we should add the self to be consistent and not confusing